### PR TITLE
Simulated Venn diagram reach by spend

### PR DESCRIPTION
**Pull request by riemanli**

### CODE REVIEW

**Review by matthewclegg — CHANGES_REQUESTED** [review]

File: src/simulator/halo_simulator.py [hunk 1]
```diff
@@ -271,28 +271,69 @@ def simulated_venn_diagram_reach_by_spend(
...
             A list of ReachPoint. Each reach point represents the mapping from
             the spends of a subset of publishers to the differentially private
             estimate of the number of people reached in this subset.
         """
-        raise NotImplementedError()
+        if max_frequency != 1:
+            raise ValueError("Max frequency has to be 1.")
+
+        venn_diagram_regions = self._form_venn_diagram_regions(spends, max_frequency)
+        # Happen when there is no active publishers, i.e. 0 spend vector.
```
[the remark this answers is not in the document]
Comment (riemanli) [reply src/simulator/halo_simulator.py:305]: Thanks for the suggestion. The comment is modified.

File: src/simulator/halo_simulator.py [hunk 2]
```diff
@@ -271,28 +271,69 @@ def simulated_venn_diagram_reach_by_spend(
...
         Args:
             spends:  The hypothetical spend vector, equal in length to
               the number of publishers.  spends[i] is the amount that is
-              spent with publisher i. Note that publishers with 0 spends will
-              not be included in the Venn diagram reach.
+              spent with publisher i.
             budget:  The amount of privacy budget that can be consumed while
               satisfying the request.
+            privacy_budget_split:  Specifies the proportion of the privacy budget
+              that should be allocated to reach estimation.  The remainder is
```
[the remark this answers is not in the document]
Comment (riemanli) [reply src/simulator/halo_simulator.py:293]: Thanks for pointing out the ambiguity. Would the following be better?
`Specifies the proportion of the privacy budget that should be allocated to the reach count in the primitive Venn diagram regions.  The remainder is allocated to the cardinality estimation from the Liquid Legions.`

File: src/simulator/halo_simulator.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (riemanli) [reply src/simulator/halo_simulator.py]: Done.

File: src/simulator/halo_simulator.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (riemanli) [reply src/simulator/halo_simulator.py]: Confirmed with Matthew and modified the docstring.

**Review by jiayu-google — APPROVED** [review]
This is awesome Rieman!

I learned much from your test.

**Review by matthewclegg — APPROVED** [review]

**Review by RonPinelliMRC — APPROVED** [review]
MRC approving based on cursory review after confirming non-independent code review on an exception basis. Experimental code will require independent review prior to publication of or reliance on results generated.

**Pull request merged**